### PR TITLE
Fix mime types

### LINKED ISSUE #25396: [Windows] test executable crashes and fails to run
### Describe the bug
Here is a clear and concise description of what the problem is:
Test executable falls to run for windows.
Git bisect found the following PR to cause

```
 c:\Users\brent\source\repos\xbmc>git bisect bad
 a0aa7baea73198e63463e7f931969c8e849a0ffa is the first bad commit
 commit a0aa7baea73198e63463e7f931969c8e849a0ffa
 Author: Arne Morten Kvarving <spiff@kodi.tv>
 Date:   Sun Feb 11 00:44:13 2024 +0100
 
     add video/VideoFileItemClassify.cpp
 
     start by moving CFileItem::IsVideo
```

However I believe newer test cases that have been introduced also are possibly causing crashes of the test executable on windows.

Jenkins was not correctly detecting a failure in the even of no output (ie a crash) from the test executable.

The following PR was made to correctly have Jenkins report such a failure case. https://github.com/xbmc/xbmc/pull/25390

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Tests to run on windows

## Actual Behavior

Test executable has no output and crashes.

Confirmed test TestVideoFileItemClassify was initial test that started crash, but I believe other tests using unix paths may also be causing crashes, as disabling the Fileitemclassify test source file still has the test executable still fail on tip of master currently.
I used a checkout of a0aa7baea73198e63463e7f931969c8e849a0ffa specifically to locate the specific test mentioned in this issue causing the crash

## Possible Fix

Definitely path related. https://github.com/xbmc/xbmc/blob/501ec1fae0fb86760548018f02702a9caf8ba997/xbmc/video/test/TestVideoFileItemClassify.cpp#L113
```
 const auto video_tests = std::array{
 //    VideoClassifyTest{"/home/user/video.avi", true, "video/avi"},
 //    VideoClassifyTest{"/home/user/video.avi", true, "", 1},
 //    VideoClassifyTest{"/home/user/video.avi", false, "", 2},
 //    VideoClassifyTest{"/home/user/video.avi", false, "", 3},
 //    VideoClassifyTest{"/home/user/video.avi", false, "", 4},
     VideoClassifyTest{"pvr://recordings/tv/1", true},
     VideoClassifyTest{"pvr://123", false},
 //    VideoClassifyTest{"dvd://VIDEO_TS/video_ts.ifo", true},
     VideoClassifyTest{"dvd://1", true},
 //    VideoClassifyTest{"/home/user/video.not", true, "application/ogg"},
 //    VideoClassifyTest{"/home/user/video.not", true, "application/mp4"},
 //    VideoClassifyTest{"/home/user/video.not", true, "application/mxf"},
 };
```
 The only ones that dont cause it to crash are the uncommented ones.

Confirmed Mimetype is not causing a failure, as well as tag and res.

### To Reproduce
1. Build kodi-test on windows. 
2. Execute kodi-test
3.no output

### Debuglog
The debuglog can be found here:
Kodi-test has no debug log. Crash is specifically caused by test case

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [X] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 22 master 501ec1fae0fb86760548018f02702a9caf8ba997

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by notspiff**
## Description
The global static data caused issues for tests since initialization order is undefined. Move to a function to 
make order deterministic. This in particular caused crashes executing the test suite on windows.

Credit goes to @neo1973 for identifying the issue. My fix is based on his proposed fix but structured slightly differently.

I also had to fix some tests that now failed to pass since the mime types are operational. This I do in a different way that actually allows the tests to test the intended code paths.

## Motivation and context
Get tests working again on windows.

## How has this been tested?
Tests pass locally..

## What is the effect on users?
None

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/video/test/TestVideoFileItemClassify.cpp [hunk 1]
```diff
@@ -113,9 +113,9 @@ TEST_P(VideoTest, IsVideo)
 const auto video_tests = std::array{
     VideoClassifyTest{"/home/user/video.avi", true, "video/avi"},
     VideoClassifyTest{"/home/user/video.avi", true, "", 1},
-    VideoClassifyTest{"/home/user/video.avi", false, "", 2},
-    VideoClassifyTest{"/home/user/video.avi", false, "", 3},
-    VideoClassifyTest{"/home/user/video.avi", false, "", 4},
+    VideoClassifyTest{"/home/user/video.gam", false, "", 2},
+    VideoClassifyTest{"/home/user/video.mus", false, "", 3},
+    VideoClassifyTest{"/home/user/video.pic", false, "", 4},
```
Comment (neo1973) [inline xbmc/video/test/TestVideoFileItemClassify.cpp:118]: I suggest to keep the existing checks and fix them and to add the new ones in addition. They check different things so it's valuable to have them both.

File: xbmc/utils/Mime.cpp [hunk 2]
```diff
@@ -18,501 +18,511 @@
...
-     {"zip", "application/zip"},
-     {"zoo", "application/octet-stream"},
-     {"zsh", "text/x-script.zsh"}}};
+namespace
+{
+
+const std::map<std::string, std::string>& GetMap()
+{
+  static const std::map<std::string, std::string> mimetypes = {
+      {{"3dm", "x-world/x-3dmf"},
```
Comment (neo1973) [inline xbmc/utils/Mime.cpp:27]: While on it I suggest to switch to `std::unordered_map` and to remove the superfluous `{}` brackets in line 27 and 520.

File: xbmc/video/test/TestVideoFileItemClassify.cpp (hunk 1 above)
Comment (notspiff) [reply xbmc/video/test/TestVideoFileItemClassify.cpp:118]: no need to have 3 duplicates of an already existing test? There is already a test that checks for mime type override. These tests are specifically there to test that tags make it return false.

File: xbmc/utils/Mime.cpp (hunk 2 above)
Comment (notspiff) [reply xbmc/utils/Mime.cpp]: done.

**Review by thexai — APPROVED** [review]

File: xbmc/utils/Mime.cpp [hunk 3]
```diff
@@ -525,8 +536,9 @@ std::string CMime::GetMimeType(const std::string &extension)
     ext = extension.substr(posNotPoint);
   transform(ext.begin(), ext.end(), ext.begin(), ::tolower);
```
Comment (ksooo) [inline xbmc/utils/Mime.cpp:537]: Could we please add `std::` so it is clear which `transform` we want to use? Maybe even change to `std::tolower`?

Comment (ksooo) [reply xbmc/utils/Mime.cpp:537]: This is just a question. The code was not touched in this PR, so completely up to you.

Comment (notspiff) [reply xbmc/utils/Mime.cpp]: std::tolower requires the stupid lambda addition. i added std::

**Comment by notspiff** [comment]
jenkins build and merge

**Comment by ksooo** [comment]
> add std:: prefix to shut up ksooo-format 

@notspiff I feel offended and will never ever talk to you from now on. :-)

**Pull request merged**